testnet/stacks-node/src/monitoring.rs

```diff
@@ -0,0 +1,51 @@
+use async_std::net::{TcpListener, TcpStream};
+use async_std::prelude::*;
+use async_std::task;
+use stacks::prometheus::{Encoder, TextEncoder, gather};
+
+pub use stacks::monitoring::{ERRORS_EMITTED_COUNTER, WARNING_EMITTED_COUNTER};
+
+use http_types::{
+    Response, 
+    StatusCode, 
+    Body,
+};
+
+pub fn start_serving_prometheus_metrics(prometheus_bind: String) {
+    let addr = prometheus_bind.clone();
+
+    async_std::task::block_on(async {
+        let listener = TcpListener::bind(addr).await.expect("todo(ludo)");
+        let addr = format!("http://{}", listener.local_addr().expect("todo(ludo)"));
+        println!("Prometheus server listening on {}", addr);
+    
+        let mut incoming = listener.incoming();
+        while let Some(stream) = incoming.next().await {
+            let stream = stream.expect("todo(ludo)");
```

Comment:
As a rule of thumb, you should never `panic!` on network code paths.  Please instead log the error and close the socket.

Reply:
And do what? loop and retry?

Reply:
Yup -- close the problematic socket and continue handling the other sockets as normal.

Reply:
Addressed with https://github.com/blockstack/stacks-blockchain/pull/1670/commits/411dc4629796bff2337875a5e3e6e9d337668b57

Reply:
I think, rather than return (shutting down the metrics thread), you should just `continue` to processing subsequent incoming connection.

Reply:
Hmm yes, addressed with https://github.com/blockstack/stacks-blockchain/pull/1670/commits/48e78f406d39d140ed0632ce6d3b9828637670c5